### ISSUE TITLE
Library empty in P9 beta, errored out on the first run

### ISSUE THREAD

**Issue by XenorPLxx** [JosefNemec/Playnite#2431]
**Before creating bug report**
- Please use issue search in the repository first. If the bug is something really obvious there's a high change it was already reported.
- Make sure that the issue is not caused by custom theme or extension. You can restart Playnite in "safe mode" from help menu to quickly test it.
- Check list of known issues https://github.com/JosefNemec/Playnite/wiki/Known-Issues

I didn't find anything.

**Integration issues**
If an issue is related to library integrations, use separate [extensions repository](https://github.com/JosefNemec/PlayniteExtensions) to file a report.

Seems like core issue.

**Describe the bug**

After making my P8 installation portable in separate folder (copy of `%localappdata%/Playnite` and then `%appdata%/Playnite` on top of that, deleting uninstallers and changing db path in config.json), deleting extensions and themes folders and unpacking P9 beta on top of that I got error saying something in lines 'cant access library because other thread owns it'. I discarder it too quickly, restarted Playnite and error didn't reappear, but also there's no library - it's completely empty.
I'm pretty sure I didn't have any other Playnite instance running and I had only main Playnite Beta folder opened up in explorer (no subfolders opened).

**To Reproduce**
If it's not clear from the description how to reproduce the issue please add clear steps to reproduce the behavior.

Not sure. I'm going to try creating new portable beta from my main installation the same way to see if it occurs again.

**Screenshots**

My hand was too fast to discard the issue.

**Diagnostics ID**
Please generate diagnostics package from "About Playnite" menu and attach identifier you received. If the issue is a crash bug, you can generate diag. package directly from crash dialog.

`7d7e8fff-0f51-4dd0-a85c-bbd1303d25b2`

**Comment by XenorPLxx** [comment]
No error on the second try, library still empty. Going to try running P8 in portable before I delete themes and extensions and overwrite with P9 files this time.

**Comment by XenorPLxx** [comment]
Different error second time around:

![image](https://user-images.githubusercontent.com/6712823/132051469-161f7126-b343-4b3b-af6d-dedab11e089e.png)
![image](https://user-images.githubusercontent.com/6712823/132051480-6fabd304-00a7-4dac-8ae6-a82d7e8c43b0.png)

`bfa522b6-9d26-4322-b9f4-97c46ec6ad20`

**Comment by XenorPLxx** [comment]
I've confirmed that making my P8 portable works fine, then updated to P9 and had the same 'Database update failed error'.

`68699fe5-193a-449b-8c59-cbccce8c98e5`

My original P8 library files:
[library.zip](https://github.com/JosefNemec/Playnite/files/7108015/library.zip)

**Comment by JosefNemec** [comment]
I can't replicate this and logs don't make much sense. This looks like you have some weird mush up of P8 and P9 files in library folder in your portable copy. Not in files you attached, because those alone can migrate just fine.

**Comment by JosefNemec** [comment]
Also that 'cant access library because other thread owns it' error is nowhere to be seen in the logs, so I have no idea what showed it, but it was not Playnite (at least it was not our code).

**Comment by XenorPLxx** [comment]
I tried removing `library/files` in P9 Beta copy, and it still throws out the same 'Database update failed' `3d9540ae-2fbc-4276-8d8b-eabafad737c8`.

If you'd like, I can upload one of the packages below containing full Playnite Portable data, although I'm not sure if there are any credentials inside:
![image](https://user-images.githubusercontent.com/6712823/132189692-9d45c589-8347-4013-8d34-cdbd20ce39c7.png)

**Comment by JosefNemec** [comment]
If you can, upload the whole directory folder after you created P9 beta folder, but BEFORE launching it for the first time.

**Comment by JosefNemec** [comment]
I forgot to mentioned that you can leave out `files` folder form library folder.

**Comment by XenorPLxx** [comment]
Here's a fresh P9, run once after making portable P8 to make sure P8 portable works, not run after upgrade to P9. `library/files` and `browsercache` removed.
https://drive.google.com/file/d/1REQRts47wteMyKYEh6MeJRtnvaM3bxtP/view?usp=sharing

After I packaged it I also tried to run it, seen same 'Database update failed error': `371e7dcf-5f97-4069-8b02-e654d098723d`

**Comment by JosefNemec** [comment]
As I suspected, you already have some P9 files there, just check `library` folder, there should be no .db files (P8 doesn't use those). You already launched P9 somewhere,, somehow and converted some files.

**Comment by XenorPLxx** [comment]
Ooookay, I think I know what I've done.

When I ran the P9 Beta for the first time, I had done it through a dedicated shortcut which had the target set up correctly, but the 'start in' path was set as my main Playnite installation folder. I guess P9 Beta created those files in the wrong folder 'thanks' to me messing up my shortcut.

That's now correctly set up and I've been using the exe file directly anyway.
![image](https://user-images.githubusercontent.com/6712823/132200810-264c7710-64f8-4f10-ad6c-12069b5f97f0.png)

And that's how my main installation now looks:
![image](https://user-images.githubusercontent.com/6712823/132200487-724fd83e-2695-4057-9deb-3b57626f1fa1.png)

Thanks to the fact I'm actually using a non-portable installation, my library was not broken (I have a copy anyway), but my P8 installation was 'polluted' with P9 files.

I've deleted 'library' folder from P8 before making it into a portable copy and then upgraded to P9 and library update was successful, after a first glance everything seems to work as expected.

I didn't reproduce the error I've mentioned in the first post 'cant access library because other thread owns it', but I guess that might've indeed a valid error - the library wasn't where Playnite was expecting it because I've made a mess with the shortcut I used.
